pkg/cliplugins/bind/plugin/bind.go

```diff
@@ -0,0 +1,174 @@
+		b.APIExportRef = args[0]
+	}
+	return nil
+}
+
+// Validate validates the BindOptions are complete and usable.
+func (b *BindOptions) Validate() error {
+	if b.APIExportRef == "" {
+		return errors.New("`root:ws:apiexport_object` reference to bind is required as an argument")
+	}
+
+	if !strings.HasPrefix(b.APIExportRef, "root") || !logicalcluster.New(b.APIExportRef).IsValid() {
+		return fmt.Errorf("fully qualified reference to workspace where APIExport exists is required. The format is `root:<ws>:<apiexport>`")
+	}
+
+	return b.Options.Validate()
+}
+
+// Run creates an apibinding for the user.
+func (b *BindOptions) Run(ctx context.Context) error {
+	config, err := b.ClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	workspacePath, apiExportName := logicalcluster.New(b.APIExportRef).Split()
+
+	// if apibindingName is not provided, default it to <apiExportname>.
+	apiBindingName := b.APIBindingName
+	if apiBindingName == "" {
+		apiBindingName = apiExportName
+	}
+
+	_, currentClusterName, err := pluginhelpers.ParseClusterURL(config.Host)
+	if err != nil {
+		return fmt.Errorf("current URL %q does not point to cluster workspace", config.Host)
+	}
+
+	binding := &apisv1alpha1.APIBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: apiBindingName,
+		},
+		Spec: apisv1alpha1.APIBindingSpec{
+			Reference: apisv1alpha1.ExportReference{
+				Workspace: &apisv1alpha1.WorkspaceExportReference{
+					Path:       workspacePath.String(),
+					ExportName: apiExportName,
+				},
+			},
+		},
+	}
+
+	kcpclient, err := newKCPClusterClient(config)
+	if err != nil {
+		return err
+	}
+
+	createdBinding, err := kcpclient.Cluster(currentClusterName).ApisV1alpha1().APIBindings().Create(ctx, binding, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	if _, err := fmt.Fprintf(b.Out, "apibinding %s created. Waiting to successfully bind ...\n", binding.Name); err != nil {
+		return err
+	}
+
+	// wait for phase to be bound
+	if createdBinding.Status.Phase != apisv1alpha1.APIBindingPhaseBound {
+		if err := wait.PollImmediate(time.Millisecond*500, b.BindWaitTimeout, func() (done bool, err error) {
+			createdBinding, err := kcpclient.Cluster(currentClusterName).ApisV1alpha1().APIBindings().Get(ctx, binding.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			if createdBinding.Status.Phase == apisv1alpha1.APIBindingPhaseBound {
+				return true, nil
+			}
+			return false, nil
+		}); err != nil {
+			return fmt.Errorf("could not bind %s: %w", binding.Name, err)
+		}
+	}
+
+	if _, err := fmt.Fprintf(b.Out, "%s created and bound.\n", binding.Name); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func newKCPClusterClient(config *rest.Config) (kcpclient.ClusterInterface, error) {
+	clusterConfig := rest.CopyConfig(config)
+	u, err := url.Parse(config.Host)
+	if err != nil {
+		return nil, err
+	}
+	u.Path = ""
+	clusterConfig.Host = u.String()
+	clusterConfig.UserAgent = rest.DefaultKubernetesUserAgent()
```

Comment:
This gets defaulted in the client-go libraries. Can we default something to show it's `kubectl-kcp-bind` plugin?

Reply:
Do we just add a prefix to this user agent? We are currently using this default for all other plugins.

Reply:
You should override it per plugin - can be a follow-up for all plugins